packages/core/contracts/financial-templates/common/financial-product-libraries/contract-for-difference-libraries/BinaryOptionContractForDifferenceFinancialProductLibrary.sol

```diff
@@ -0,0 +1,67 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/utils/math/SafeMath.sol";
+
+import "./ContractForDifferenceFinancialProductLibrary.sol";
+import "../../../../common/implementation/Lockable.sol";
+
+/**
+ * @title Binary Option Contract For Difference Financial Product Library.
+ * @notice Adds settlement logic to simply binary option CFDs. Binary options settle with all collateral allocated to
+ * either the long or short side, depending on the settlement price. They can be used to make prediction markets or any
+ * kind of binary bet. Settlement is defined using a strike price which informs which side of the bet was correct. If
+ * settlement price is greater or equal to the strike then all value is sent to the long side. Otherwise, all value
+ * is sent to the short side. The settlement price could be a scalar (like the price of ETH) or a binary bet with
+ * settlement being 0 or 1 depending on the outcome.
+ */
+contract BinaryOptionContractForDifferenceFinancialProductLibrary is
+    ContractForDifferenceFinancialProductLibrary,
+    Lockable
+{
+    using FixedPoint for FixedPoint.Unsigned;
+    using SafeMath for uint256;
```

Comment:
is safemath needed in 0.8?

Reply:
no but we import it everywhere else! I don't think it adds much gas as the OZ lib simply does unchecked operations under the hood and relies on the solidity validation